### ISSUE TITLE
use-strict rule doesn't check IIFE lambda

### ISSUE THREAD

**Issue by pixelshaded** [palantir/tslint#458]
I'm linting this file:

```ts
(() => {
  angular.module('app', [
    'ngSanitize',
    'ngAnimate',
    'ui.router',
    'ui.bootstrap',
    'app.config',
    'app.routes',
    'app.services',
    'app.factories',
    'app.controllers',
    'app.directives',
    'companies',
    'features',
    'users',
    'roles'
  ]);
})();
```

With these rules:

```json
"use-strict": [true,
      "check-module",
      "check-function"
    ],
```

And the linter is passing. Shouldn't check-function require that use strict be inside the lambda body?

Note that I am using gulp-tslint (which is using tslint ~2.2.0-beta) in case that matters.

**Comment by adidahiya** [comment]
From the readme:

> * `use-strict` enforces ECMAScript 5's strict mode
>     * `check-module` checks that all top-level modules are using strict mode
>     * `check-function` checks that all top-level functions are using strict mode

`check-function` only looks at top-level function declarations, not lambdas like that. If you're compiling TS to a module system (commonjs or AMD), you shouldn't have to include that IIFE. Otherwise, I'd suggest using internal modules / namespaces:

```ts
module MyAppNamespace {
    angular.module('app', [
        ...
    ]);
}
```

which gets compiled to:

```js
var MyAppNamespace;
(function (MyAppNamespace) {
    angular.module('app', [
        ...
    ]);
})(MyAppNamespace || (MyAppNamespace = {}));
```

... thereby accomplishing the same thing while being more TS-idiomatic.

**Comment by pixelshaded** [comment]
I guess the moral of the story is to read. I think I made some assumptions on how it is checking for use strict. Some other linters make sure your code is in strict mode based on use strict hierarchy. Thanks for the info.
